packages/flutter_tools/lib/src/build_system/targets/localizations.dart

```diff
@@ -0,0 +1,285 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+import 'package:process/process.dart';
+import 'package:yaml/yaml.dart';
+
+import '../../artifacts.dart';
+import '../../base/file_system.dart';
+import '../../base/io.dart';
+import '../../base/logger.dart';
+import '../../globals.dart' as globals;
+import '../build_system.dart';
+import '../depfile.dart';
+
+/// Run the localizations generation script with the configuration [options].
+Future<Depfile> generateLocalizations({
+  @required LocalizationOptions options,
+  @required String flutterRoot,
+  @required FileSystem fileSystem,
+  @required ProcessManager processManager,
+  @required Logger logger,
+  @required Directory projectDir,
+  @required String dartBinaryPath,
+}) async {
+  // Setup project inputs and outputs. This currently makes some
+  // guess about the output and does not include all files.
+  final Directory inputArb = fileSystem.directory(
+    options.arbDirectory ?? projectDir
+      .childDirectory('lib')
+      .childFile('l10n').uri,
+  );
+
+  File outputLocalzations;
+  if (options.outputLocalizationsFile != null) {
+    outputLocalzations = fileSystem
+      .file(inputArb.uri.resolveUri(options.outputLocalizationsFile));
+  } else {
+    outputLocalzations = projectDir
+    .childDirectory('lib')
+    .childFile('app_localizations.dart');
+  }
+
+  if (!inputArb.existsSync()) {
+    logger.printError('arb-dir: ${inputArb.path} does not exist.');
+    throw Exception();
```

Comment:
I don't quite remember how/whether the build system aggregates errors, but I sort of remember that exceptions from running a target will get printed at some point, and at that point will it be confusing to have one with an empty message?

Reply:
What this looks like is:

```
arb-dir: /Users/jonahwilliams/Documents/flutter/examples/hello_world/lib/l10n does not exist.
Target gen_localizations failed: Exception                              
Compiling lib/main.dart for the Web...                              0.1s
Exception: Failed to compile application for the Web.
```


I think its fine, we show the error text and that is the important part. We can't control the formatting as easily if we stick it in the exception.